worker/s3caller/client.go

```diff
@@ -79,6 +79,14 @@ func (l *awsLogger) Logf(classification logging.Classification, format string, v
 	}
 }
 
+type NoOpRateLimit struct{}
```

Comment:
Do the types really need to be exported? If they do, they should have comments.

I'd allude to "unlimited" or "no limit" in the name rather than "no op".